auto-update.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Keeps wc-api-dev up to date based on GitHub releases.
+ */
+class WC_API_Dev_Updater {
+
+	// Plugin file path / slug
+	protected $file = 'wc-api-dev/wc-api-dev.php';
+	protected $github_response = null;
+
+	/**
+	 * Hooks into the plugin update system.
+	 */
+	public function __construct() {
+		add_filter( 'pre_set_site_transient_update_plugins', array( $this, 'modify_transient' ), 10, 1 );
+		add_filter( 'plugins_api', array( $this, 'plugin_popup' ), 10, 3 );
+		add_filter( 'auto_update_plugin', array( $this, 'auto_update' ), 10, 2 );
+		add_filter( 'upgrader_post_install', array( $this, 'after_install' ), 10, 3 );
+	}
+
+	/**
+	 * Adds the wc-api-dev plugin to the list of allowed plugins for auto update.
+	 */
+	function auto_update( $update, $item ) {
+		if ( in_array( $item->slug, array( 'wc-api-dev' ) ) ) {
+			return true;
+		} else {
+			return $update;
+		}
+	}
+
+	/**
+	 * Fetch the first non pre-release zip from GitHub releases and store the response for later use.
+	 */
+	private function maybe_fetch_github_response() {
+		if ( is_null( $this->github_response ) ) {
+			$request_uri = 'https://api.github.com/repos/woocommerce/wc-api-dev/releases';
+			$response = json_decode( wp_remote_retrieve_body( wp_remote_get( $request_uri ) ), true );
+			if ( is_array( $response ) ) {
+				foreach( $response as $entry ) {
+					if ( false === ( bool ) $entry['prerelease'] ) {
+						$this->github_response = $entry;
+						break;
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Add our plugin to the list of plugins to update, if we find the version is out of date.
+	 */
+	public function modify_transient( $transient ) {
+		if( property_exists( $transient, 'checked' ) && $transient->checked ) {
+			$checked = $transient->checked;
+			$this->maybe_fetch_github_response();
+			$out_of_date = version_compare( $this->github_response['tag_name'], $checked[ $this->file ], 'gt' );
```

Comment:
We should check `github_response` for `'tag_name'` and `$checked` for `$this->file` to avoid warnings.

Nitpick: WP core uses '>' for `version_compare()` calls.